src/App.js

```diff
@@ -1,25 +1,48 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from 'react'
+import Welcome from './components/Welcome.js'
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      error: null,
+      isLoaded: false,
+      welcomeString: 'hello'
+    }
+  }
+
+  componentDidMount () {
+    fetch('https://safe-wave-09726.herokuapp.com/api/v1/welcome')
```

Comment:
Part of the reason that this is difficult to test is that you have a `fetch` directly in your componentDidMount. So every test you write that sets up `App.js` is literally going to make a GET request to your API. You might want to consider making an "Api" module (or something like that) which handles calls to the Api and returns results / errors. Then, you can mock that Api when you perform `componentDidMount`

Reply:
This is EXTREMELY helpful. I saw this along the way, I will implement it.